### ISSUE TITLE
IDISP001 should warn here

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#54]
```cs
namespace RoslynSandbox
{
    using System.IO;

    public class Foo
    {
        public Foo()
        {
            Stream stream;
            if (TryGetStream(out stream))
            {
            }
        }

        private static bool TryGetStream(out Stream stream)
        {
            stream = File.OpenRead(string.Empty);
            return true;
        }
    }
}
```

And C# 7
```cs
namespace RoslynSandbox
{
    using System.IO;

    public class Foo
    {
        public Foo()
        {
            if (TryGetStream(out var stream))
            {
            }
        }

        private static bool TryGetStream(out Stream stream)
        {
            stream = File.OpenRead(string.Empty);
            return true;
        }
    }
}
```

**Comment by JohanLarsson** [comment]
https://github.com/DotNetAnalyzers/IDisposableAnalyzers/commit/ae2cb35479510ae8df66826ec2b01e4987338123
